learning_journal_basic1/tests.py

```diff
@@ -1,29 +1,162 @@
-import unittest
+import pytest
+import transaction
+import datetime
+from ..views.default import TIME_FORMAT
 
 from pyramid import testing
 
+from .models import (
+    MyModel,
+    get_engine,
+    get_session_factory,
+    get_tm_session,
+)
+from .models.meta import Base
 
-class ViewTests(unittest.TestCase):
-    def setUp(self):
-        self.config = testing.setUp()
+DB_SETTINGS = {'sqlalchemy.url': 'sqlite:////tmp/testme.sqlite'}
 
-    def tearDown(self):
+
+@pytest.fixture(scope="session")
+def sqlengine(request):
+    """Set up sql engine for testing."""
+    config = testing.setUp(settings=DB_SETTINGS)
+    config.include(".models")
+    settings = config.get_settings()
+    engine = get_engine(settings)
+    Base.metadata.create_all(engine)
+
+    def teardown():
         testing.tearDown()
+        transaction.abort()
+        Base.metadata.drop_all(engine)
+
+    request.addfinalizer(teardown)
+    return engine
+
+
+@pytest.fixture(scope="function")
+def new_session(sqlengine, request):
+    """Set up new session for testing."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    def teardown():
+        transaction.abort()
+
+    request.addfinalizer(teardown)
+    return session
+
+
+@pytest.fixture(scope="function")
+def populated_db(request, sqlengine):
+    """Set up mock database for testing."""
+    session_factory = get_session_factory(sqlengine)
+    session = get_tm_session(session_factory, transaction.manager)
+
+    with transaction.manager:
+        session.add(MyModel(title="Purgatory", body="The reality of pythonic purgatory cannot be denied.", date=datetime.datetime.utcnow()))
+
+    def teardown():
+        with transaction.manager:
+            session.query(MyModel).delete()
+
+    request.addfinalizer(teardown)
+
+
+@pytest.fixture()
+def dummy_request(new_session):
+    """Testing a dummy request."""
+    return testing.DummyRequest(dbsession=new_session)
+
+
+def test_home_view(dummy_request, new_session):
+    """Test entries are in home view."""
+    from .views.default import home_view
+    new_session.add(MyModel(title="test", body="this is a test", date=datetime.strptime(entry['date'], TIME_FORMAT)))
+    new_session.flush()
+    info = home_view(dummy_request)
+    assert "entries" in info
+
+
+def test_detail_view(new_session):
+    """Test detail view has a body."""
+    from .views.default import detail_view
+    request = testing.DummyRequest(dbsession=new_session)
+    new_session.add(MyModel(title="test", body="this is a test", date=datetime.strptime(entry['date'], TIME_FORMAT)))
+    new_session.flush()
+    request.matchdict = {'id': '1'}
```

Comment:
How certain are you that the ID of your new entry is `1`?  Might it be something else?  Can you find the ID of your new entry without guessing, or hoping it will be a stable, repeatable value?